### PR TITLE
Issue 987 - Add detection_threshold arg for all Lightning adapters

### PR DESCRIPTION
**Pull request by Anjum48**
Fix for issue #987. Any feedback is appreciated! 😀️

The only bit I wasn't sure about was the `convert_raw_predictions` in the `mmdet` adapter (the `import *` style makes tracing dependence tricky). I had to remove `self.` to make it work but I'm not sure if that would break other things.

### CODE REVIEW

**Comment by FraPochetti** [comment]
Hi @Anjum48, thanks for your contribution!
I see some tests are failing.
Is everything green on your local machine?

**Comment by Anjum48** [comment]
Hi @FraPochetti, Yes I'm seeing failed tests locally too. I'll try and track them down and get back to you if I need some help

**Comment by Anjum48** [comment]
All tests passing on my side so 🤞️

**Comment by FraPochetti** [comment]
Nice! All tests are passing. We will review ASAP.

File: icevision/models/mmdet/lightning/model_adapter.py [hunk 1]
```diff
@@ -59,16 +66,16 @@ def validation_step(self, batch, batch_idx):
...
+            records=records,
+            detection_threshold=self.detection_threshold,
         )
         self.accumulate_metrics(preds)
 
         for k, v in outputs["log_vars"].items():
             self.log(f"valid/{k}", v)
 
-        # TODO: is train and eval model automatically set by lighnting?
-        self.model.train()
```
Comment (FraPochetti) [inline icevision/models/mmdet/lightning/model_adapter.py:71]: Why did you remove this line?

Comment (Anjum48) [reply icevision/models/mmdet/lightning/model_adapter.py:71]: PyTorch Lightning automatically takes care of `model.train()` and `model.eval()`. I removed it to make it consistent with the other lightning adapters

**Comment by fstroth** [comment]
Looks all good to me! The only suggestion I would have would be to change the parameter name to `validation_detection_threshold` to indicate the intention/usage of the threshold. Should make it easier to understand the code when reading it for the first time or when people check the init parameters.

File: icevision/models/mmdet/lightning/model_adapter.py (hunk 1 above)
Comment (FraPochetti) [reply icevision/models/mmdet/lightning/model_adapter.py:71]: @potipot do you recall why this was there in the first place?
I think @Anjum48 point is valid.

**Comment by FraPochetti** [comment]
> Looks all good to me! The only suggestion I would have would be to change the parameter name to `validation_detection_threshold` to indicate the intention/usage of the threshold. Should make it easier to understand the code when reading it for the first time or when people check the init parameters.

@Anjum48 would you mind implementing @fstroth suggestion, please?

**Comment by potipot** [comment]
I think `validation_confidence_threshold` might be even more accuratem WDYT?

**Comment by FraPochetti** [comment]
> I think `validation_confidence_threshold` might be even more accuratem WDYT?

+ 1
